### PR TITLE
fix Kafka flaky test

### PR DESCRIPTION
**Pull request by Roiocam**
Trying to solve #103, I am very doubtful that KafakConsumer will not be consumed evenly from two partitions.

### CODE REVIEW

**Comment by Roiocam** [comment]
After four times CI checks(see: https://github.com/apache/incubator-pekko-projection/actions/runs/7831923269), this kind of test has been successfully passed, and we should be able to assume that this kind of flaky has been fixed.

**Comment by Roiocam** [comment]
@raboof Could you please do a code review? Thanks.

**Review by raboof — COMMENTED** [review]
TBH I'm not convinced this change fixes the root cause of the instability: because of the assertion `records.count(_.partition() == tp0.partition()) shouldBe 5` AFAICS it should be guaranteed that `tp0TestCount` would always be 5, so this change should not have an effect other than timing differences - or did I miss something?

**Comment by Roiocam** [comment]
> TBH I'm not convinced this change fixes the root cause of the instability: because of the assertion `records.count(_.partition() == tp0.partition()) shouldBe 5` AFAICS it should be guaranteed that `tp0TestCount` would always be 5, so this change should not have an effect other than timing differences - or did I miss something?

The producer produces a total of 20 elements, and each partition has 10 evenly.

Then Sink consumes 10 elements from two partitions. There is no assertion that each partition consumes 5 evenly. If it is not consumed evenly, it will lead to problems with subsequent assertions, because partition 0 may consume 6 records in the previous step.

This PR does not force the assertion that Sink consumes evenly from all partitions. If Kafka Consumer is not balanced consume from partitions, I don't think this is the responsibility of this test or should be added a new test to assert.

File: kafka-test/src/test/scala/org/apache/pekko/projection/kafka/internal/KafkaSourceProviderImplSpec.scala [hunk 1]
```diff
@@ -109,6 +110,12 @@ class KafkaSourceProviderImplSpec extends ScalaTestWithActorTestKit with LogCapt
           records.count(_.partition() == tp1.partition()) shouldBe 5
         }
 
+        // because source push to handle(probe) before sinkProbe request pull, it made probe cache random one record
+        val eagerMessage = probe.receiveMessage()
```
Comment (Roiocam) [inline kafka-test/src/test/scala/org/apache/pekko/projection/kafka/internal/KafkaSourceProviderImplSpec.scala:114]: It's kind of like https://github.com/akka/akka-projection/issues/462, i won't say this is best solution.

Comment (Roiocam) [reply kafka-test/src/test/scala/org/apache/pekko/projection/kafka/internal/KafkaSourceProviderImplSpec.scala:114]: @He-Pin Is there any way to make the source and sink reach a consensus on the number of elements?

**Comment by Roiocam** [comment]
Would you like to take a look? @raboof 

This may not the best solution, but it does reduce flaky.

**Review by raboof — APPROVED** [review]
OK, not sure I understand why it would, but if it does that would be awesome :)

**Pull request merged**